lpcompiler.pas

```diff
@@ -2561,6 +2578,28 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
     end;
   end;
 
+  function __IsProperty(typ: TLapeType): Boolean;
+  begin
+    Result := (Typ <> nil) and (Typ is TLapeType_OverloadedMethod) and (TLapeType_OverloadedMethod(Typ).MethodDef = mdProperty);
+  end;
+
+  function IsProperty(Node: TLapeTree_ExprBase): Boolean;
+  begin
+    Result := False;
+    if TLapeTree_Base.isEmpty(Node) or (not (Node is TLapeTree_ExprBase)) or (Node is TLapeTree_Invoke) then
+      Exit();
+
+    if __IsProperty(Node.resType()) then
+      Exit(True);
+
+    if (Node is TLapeTree_Operator) then
```

Comment:
Yes, but we need to recurse anyway. Either here or in `ResolveMethods`. I would prefer to do it only once.

Previously, we could skip the recursion step using a simple options check (e.g. `AutoInvoke`). Now (see line 2653), it's probably better to do the check on the fly.